parser/transaction.go

```diff
@@ -388,7 +388,7 @@ func (tx *Transaction) OrchardActionsCount() int {
 func (tx *Transaction) ToCompact(index int) *walletrpc.CompactTx {
 	ctx := &walletrpc.CompactTx{
 		Index: uint64(index), // index is contextual
-		Hash:  hash32.ToSlice(tx.GetEncodableHash()),
+		Txid:  hash32.ToSlice(tx.GetEncodableHash()),
```

Comment:
Aside: While checking that `tx.GetEncodableHash()` returns the correct data, I saw that `tx.GetDisplayHash()` also exists. This method should not exist; it should be deleted, and its contents inlined into `tx.GetDisplayHashString()`.